pkg/security/tests/activity_dumps_test.go

```diff
@@ -229,15 +231,16 @@ func TestActivityDumps(t *testing.T) {
 
 		time.Sleep(2 * time.Second) // a quick sleep to let starts and snapshot events to be added to the dump
 
-		for i := 0; i < 20; i++ {
+		for i := 0; i < testActivityDumpRateLimiter*10; i++ {
 			if err := os.MkdirAll("/tmp/ratelimiter", os.ModePerm); err != nil {
 				t.Fatal(err)
 			}
+			defer os.Remove("/tmp/ratelimiter")
```

Comment:
The defer is executed at the end of the function, so we are pushing `testActivityDumpRateLimiter*10 ` remove of the same directory to the end of the function here. Maybe we can use `t.TempDir` https://pkg.go.dev/testing#T.TempDir ?

Reply:
You're right .. Now it should be better, thank you!